frontend/javascripts/dashboard/explorative_annotations_view.tsx

```diff
@@ -324,24 +378,7 @@ class ExplorativeAnnotationsView extends React.PureComponent<Props, State> {
   };
 
   renameTracing(tracing: APIAnnotationInfo, name: string) {
-    const tracings = this.getCurrentTracings();
-    const newTracings = tracings.map((currentTracing) => {
-      if (currentTracing.id !== tracing.id) {
-        return currentTracing;
-      } else {
-        return update(currentTracing, {
-          name: {
-            $set: name,
-          },
-        });
-      }
-    });
-    this.setModeState(
-      {
-        tracings: newTracings,
-      },
-      this.state.shouldShowArchivedTracings,
-    );
+    this.updateTracingInLocalState(tracing, (t) => update(t, { name: { $set: name } }));
     editAnnotation(tracing.id, tracing.typ, {
```

Comment:
could you also catch the error case here in case the update didn't work? i.e., show an error toast and restore the name again. maybe it makes sense to switch the order, too (first update on server, then update locally; however I'm not sure if this causes a flicker or something similar).

Reply:
I decided to first update locally so the user does not get confused why the name not changed successful when having something of a connection bottleneck. In case of an error I shown an error toast and revert back to the previous name (so the UI should by in sync with the data saved by the backend)

Reply:
Nvm this contradicts how I implemented `setLockedState`. Thus, I first update the backend and if that worked out, update the view in the frontend